nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -2283,7 +2285,21 @@ private boolean mayBeNullExpr(VisitorState state, ExpressionTree expr) {
         // obviously null
         return true;
       case ARRAY_ACCESS:
-        // unsound!  we cannot check for nullness of array contents yet
+        if (config.isJSpecifyMode()) {
+          ArrayAccessTree arrayAccess = (ArrayAccessTree) expr;
+          ExpressionTree arrayExpr = arrayAccess.getExpression();
+          Symbol arraySymbol = ASTHelpers.getSymbol(arrayExpr);
+          if (arraySymbol != null) {
+            for (Attribute.TypeCompound t : arraySymbol.getRawTypeAttributes()) {
+              for (TypeAnnotationPosition.TypePathEntry entry : t.position.location) {
+                if (entry.tag == TypeAnnotationPosition.TypePathEntryKind.ARRAY) {
+                  return true;
+                }
+              }
```

Comment:
This is a good start!  I think we want to extend this logic, as follows.  Move the `ARRAY_ACCESS` case down under this switch: 

https://github.com/uber/NullAway/blob/5fbee1ff8682762cd2f291070491fe3e476787cf/nullaway/src/main/java/com/uber/nullaway/NullAway.java#L2343

The logic should be just to set `exprMayBeNull` to true iff the type of the array contents is `@Nullable` (similar to the logic you have already written).

Then, we need to update the dataflow logic as well; that logic is here:

https://github.com/uber/NullAway/blob/master/nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java#L794-L794

For now, rather than just using `defaultAssumption` we can again set the nullability of the expression based on the type of the array contents.    So, the logic you wrote already should be extracted to a utility method that can be called from both places.  Eventually, we want to extend the dataflow logic to handle null checks, so code like `if (fizz[0] != null) { fizz[0].length(); }` does not issue any warnings.  But that doesn't need to happen in this PR.